project/cmake/modules/FindCXX11.cmake

```diff
@@ -0,0 +1,18 @@
+include(TestCXXAcceptsFlag)
+
+# try to use compiler flag -std=c++11
+check_cxx_accepts_flag("-std=c++11" CXX_FLAG_CXX11)
+if(CXX_FLAG_CXX11)
+  add_options (CXX ALL_BUILDS "-std=c++11")
+  set(CXX_STD11_FLAGS "-std=c++11")
+else()
```

Comment:
Starting with CMake 3.1 we could simply use this in the main CMakeLists.txt which works [cross platform and for all interesting compilers](https://cmake.org/cmake/help/latest/manual/cmake-compile-features.7.html#supported-compilers). If there are no objections, I'd go for this. 
```
set(CMAKE_CXX_STANDARD 11)
set(CMAKE_CXX_STANDARD_REQUIRED ON)
set(CMAKE_CXX_EXTENSIONS OFF)  # Strict standard, no compiler specific extensions (such as gnu11)
```